<b>Base de données accidents corporels de la circulation</b>

Pour cet exercice nous allons utiliser quelques données tabulaires (format CSV) sur les accidents corporels liés de la circulation https://www.data.gouv.fr/fr/datasets/bases-de-donnees-annuelles-des-accidents-corporels-de-la-circulation-routiere-annees-de-2005-a-2020/#description.

<li>Importer les tableaux de données dans trois dataframes différents </li>

In [2]:
import numpy as np
import pandas as pd
caracter= pd.read_csv("caracteristiques-2017.csv", delimiter=',', encoding='ISO-8859-1', dtype='str')
#suite UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe9 in position 862: invalid continuation byte, j'ai ajouté le (encoding='ISO-8859-1') 
caracter
#caracter.info()

Num_Acc  an mois jour  hrmn lum agg int atm col  com  \
0      201700000001  17    1   11  1820   5   2   1   1   1  477   
1      201700000002  17    2   13  1630   1   2   3   1   3  005   
2      201700000003  17    3    7  1150   1   2   9   1   5  052   
3      201700000004  17    4   22  1300   1   2   1   1   6  005   
4      201700000005  17    5   20  1230   1   2   1   1   2  011   
...             ...  ..  ...  ...   ...  ..  ..  ..  ..  ..  ...   
60696  201700060697  17   11   19  1030   1   2   1   1   6  611   
60697  201700060698  17   11   25   624   1   1   6   1   7  611   
60698  201700060699  17   11   26  1745   2   2   1   1   3  611   
60699  201700060700  17   11   28  1405   1   2   1   1   3  611   
60700  201700060701  17   11   30  1939   3   2   1   1   6  611   

                            adr gps      lat     long  dep  
0                 rue nationale   M  5051326  0292191  590  
1              5 rue sonneville   M  5053611  0295314  590  
2              rue Jules Guesde   M  5052174  0288786  590  
3             46 rue Sonneville   M  5053723  0295700  590  
4            Rue roger salengro   M  5052999  0293798  590  
...                         ...  ..      ...      ...  ...  
60696  1, tsoundzou-Baraka (rue   Y  1280703  4520703  976  
60697  1, mamoudzou centre-Adri   Y  1278296  4523434  976  
60698  1, tsoundzou-Abdallah So   Y  1280711  4520617  976  
60699   1, kaweni-Kaweni nord (   Y  1276210  4522609  976  
60700     tsoundzou-Stara (rue)   Y  1281342  4520169  976  

[60701 rows x 16 columns]

In [3]:
lieu= pd.read_csv("lieux-2017.csv", delimiter=',', dtype='str')
lieu
lieu.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60701 entries, 0 to 60700
Data columns (total 18 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Num_Acc  60701 non-null  object
 1   catr     60701 non-null  object
 2   voie     50915 non-null  object
 3   v1       408 non-null    object
 4   v2       2694 non-null   object
 5   circ     60330 non-null  object
 6   nbv      60267 non-null  object
 7   pr       26207 non-null  object
 8   pr1      25968 non-null  object
 9   vosp     60105 non-null  object
 10  prof     60252 non-null  object
 11  plan     59916 non-null  object
 12  lartpc   58538 non-null  object
 13  larrout  58681 non-null  object
 14  surf     60236 non-null  object
 15  infra    57037 non-null  object
 16  situ     57227 non-null  object
 17  env1     56996 non-null  object
dtypes: object(18)
memory usage: 8.3+ MB


In [4]:
user= pd.read_csv("usagers-2017.csv", delimiter=',', dtype='str')
user
#user.info()

Num_Acc place catu grav sexe trajet secu locp actp etatp an_nais  \
0       201700000001     1    1    3    1      9   13    0    0     0    1968   
1       201700000001     2    2    3    2      9   11    0    0     0    1973   
2       201700000001     1    1    3    1      1   13    0    0     0    1967   
3       201700000002     1    1    1    1      0   11    0    0     0    1953   
4       201700000002     1    1    3    1      5   22    0    0     0    1960   
...              ...   ...  ...  ...  ...    ...  ...  ...  ...   ...     ...   
136016  201700060699     1    1    1    2      9   11    0    0     0    1974   
136017  201700060700     1    1    1    2      9   11    0    0     0    1987   
136018  201700060700     1    1    4    1      9   21    0    0     0    1991   
136019  201700060700     2    2    4    2      9   21    0    0     0    1990   
136020  201700060701     1    1    4    1      1   21    0    0     0    1992   

       num_veh  
0          B01  
1          B01  
2          A01  
3          A01  
4          B01  
...        ...  
136016     B01  
136017     A01  
136018     B01  
136019     B01  
136020     A01  

[136021 rows x 12 columns]

- Combien y a t-il d'observations dans ces datasets ? y a t-il des valeurs manquantes ? 
    - caracter: 60701 rows × 16 columns, donc 60701 d'observations dans cette dataset
    - lieu: 60701 rows × 18 columns, donc 60701 d'observations dans cette dataset
    - user:136021 rows × 12 columns, donc 136021 d'observations dans cette dataset
    - oui, caracter.info() :
        - 11  adr      59879 non-null  object
        - 12  gps      56093 non-null  object
        - 13  lat      52970 non-null  object
        - 14  long     52970 non-null  object
    - oui, lieu.info() :
        - 1   catr     60701 non-null  object
        - 2   voie     50915 non-null  object
        - 3   v1       408 non-null    object
        - 4   v2       2694 non-null   object
        - 5   circ     60330 non-null  object
        - 6   nbv      60267 non-null  object
        - 7   pr       26207 non-null  object
        - 8   pr1      25968 non-null  object
        - 9   vosp     60105 non-null  object
        - 10  prof     60252 non-null  object
        - 11  plan     59916 non-null  object
        - 12  lartpc   58538 non-null  object
        - 13  larrout  58681 non-null  object
        - 14  surf     60236 non-null  object
        - 15  infra    57037 non-null  object
        - 16  situ     57227 non-null  object
        - 17  env1     56996 non-null  object
    - oui, user.info() :
       - RangeIndex: 136021 entries, 0 to 136020
       - Data columns (total 12 columns):
       ---  ------   --------------   ----- 
                - Column   Non-Null Count   Dtype 
            - 0   Num_Acc  136021 non-null  object
            - 1   place    124219 non-null  object
            - 2   catu     136021 non-null  object
            - 3   grav     136021 non-null  object
            - 4   sexe     136021 non-null  object
            - 5   trajet   136010 non-null  object
            - 6   secu     127071 non-null  object
            - 7   locp     135975 non-null  object
            - 8   actp     135978 non-null  object
            - 9   etatp    135955 non-null  object
            - 10  an_nais  135984 non-null  object
            - 11  num_veh  136021 non-null  object
- Combien de colonnes dans caracteristiques2017 ?
    - Data columns (total 16 columns):
- Quels sont les types des colonnes de usagers2017 et caracteristiques2017 ? 
    - caracteristiques2017: dtypes: object(16)
    - usagers2017: dtypes: object(12)
- Combien y a t-il de doublon dans la colonne Num_acc dans le dataframe usager2017 ?
- Combien y a t-il de doublon dans la colonne Num_acc dans le dataframe caracteristique2017 ?
- C'est quoi le type de la relation si nous voulons concaténer caracteristique2017 et usagers2017 sur la colone num_acc?

In [5]:
#Combien y a t-il de doublon dans la colonne Num_acc dans le dataframe usager2017 ?
user.duplicated().sum() #==>143
user.duplicated()

0         False
1         False
2         False
3         False
4         False
          ...  
136016    False
136017    False
136018    False
136019    False
136020    False
Length: 136021, dtype: bool

In [6]:
#Combien y a t-il de doublon dans la colonne Num_acc dans le dataframe caracterstique2017 ?
caracter.duplicated().sum() #==>0

0

 <li>Remplacer toutes les valeurs 1 et 2 dans la colonne sexe par "homme" et "femme"</li>

In [7]:
#  user['sexe'] est un objet, donc la valeur 1 il faut mettre dans un '' à la fin de faire la comparison,
# mais si je voulais utiliser le 1 et 2 comme de chiffres je dois convertir le type de colonne en entier
# et puis je fais la comparison : user.loc[(user['sexe'] == 1), 'sexe'] = 'homme' 
user.loc[(user['sexe'] == '1'), 'sexe'] = 'homme' 
user.loc[(user['sexe'] == '2'), 'sexe'] = "femme" 
#user['sexe']=user['sexe'].map({'1':'homme', '2':'femme'})
user

Num_Acc place catu grav   sexe trajet secu locp actp etatp  \
0       201700000001     1    1    3  homme      9   13    0    0     0   
1       201700000001     2    2    3  femme      9   11    0    0     0   
2       201700000001     1    1    3  homme      1   13    0    0     0   
3       201700000002     1    1    1  homme      0   11    0    0     0   
4       201700000002     1    1    3  homme      5   22    0    0     0   
...              ...   ...  ...  ...    ...    ...  ...  ...  ...   ...   
136016  201700060699     1    1    1  femme      9   11    0    0     0   
136017  201700060700     1    1    1  femme      9   11    0    0     0   
136018  201700060700     1    1    4  homme      9   21    0    0     0   
136019  201700060700     2    2    4  femme      9   21    0    0     0   
136020  201700060701     1    1    4  homme      1   21    0    0     0   

       an_nais num_veh  
0         1968     B01  
1         1973     B01  
2         1967     A01  
3         1953     A01  
4         1960     B01  
...        ...     ...  
136016    1974     B01  
136017    1987     A01  
136018    1991     B01  
136019    1990     B01  
136020    1992     A01  

[136021 rows x 12 columns]

   <li>Afficher les femmes qui ont subi des accidents</li>

In [8]:
user.loc[(user['sexe'] == 'femme'), ['Num_Acc','sexe']]
#user['Num_Acc'].isnull

Num_Acc   sexe
1       201700000001  femme
8       201700000003  femme
9       201700000004  femme
15      201700000008  femme
18      201700000009  femme
...              ...    ...
136010  201700060696  femme
136013  201700060697  femme
136016  201700060699  femme
136017  201700060700  femme
136019  201700060700  femme

[43937 rows x 2 columns]

   <li>Remplacez chaque entier de la colonne grav (gravité de l'accident) par leur mappage correspondant : (1)Indemne (2)Tué (3)Blessé hospitalsé (4)Blessé léger</li>
    <li>Afficher le nombre d'occurence de chaque valeur dans lacolonne grav</li>

In [9]:
user.loc[(user['grav']== '1'), 'grav'] = 'Indemne'
user.loc[(user['grav']== '2'), 'grav'] = 'Tué'
user.loc[(user['grav']== '3'), 'grav'] = 'Blessé hospitalsé'
user.loc[(user['grav']== '4'), 'grav'] = 'Blessé léger'
user

Num_Acc place catu               grav   sexe trajet secu locp  \
0       201700000001     1    1  Blessé hospitalsé  homme      9   13    0   
1       201700000001     2    2  Blessé hospitalsé  femme      9   11    0   
2       201700000001     1    1  Blessé hospitalsé  homme      1   13    0   
3       201700000002     1    1            Indemne  homme      0   11    0   
4       201700000002     1    1  Blessé hospitalsé  homme      5   22    0   
...              ...   ...  ...                ...    ...    ...  ...  ...   
136016  201700060699     1    1            Indemne  femme      9   11    0   
136017  201700060700     1    1            Indemne  femme      9   11    0   
136018  201700060700     1    1       Blessé léger  homme      9   21    0   
136019  201700060700     2    2       Blessé léger  femme      9   21    0   
136020  201700060701     1    1       Blessé léger  homme      1   21    0   

       actp etatp an_nais num_veh  
0         0     0    1968     B01  
1         0     0    1973     B01  
2         0     0    1967     A01  
3         0     0    1953     A01  
4         0     0    1960     B01  
...     ...   ...     ...     ...  
136016    0     0    1974     B01  
136017    0     0    1987     A01  
136018    0     0    1991     B01  
136019    0     0    1990     B01  
136020    0     0    1992     A01  

[136021 rows x 12 columns]

In [10]:
user['grav'].value_counts()
#OU pour chaque valeur séparée
#user.loc[user['grav']== 'Indemne','grav'].count() #==>56270
#user.loc[user['grav']== 'Blessé léger','grav'].count() #==> 47158
#user.loc[user['grav']== 'Blessé hospitalsé','grav'].count() #==> 28993
#user.loc[user['grav']== 'Tué','grav'].count() #==> 3600

Indemne              56270
Blessé léger         47158
Blessé hospitalsé    28993
Tué                   3600
Name: grav, dtype: int64

   <li>Remplacez chaque entier de la colonne catu (catégorie d'usagers) par leur mappage correspondant : (1)Conducteur (2)Passager (3)Piéton (4)Piéton en roller ou en trotinette (5) Autre Véhicule</li>

In [11]:
user['catu']= user['catu'].map({'1':'Conducteur', '2': 'Passager', '3': 'Piéton', '4': 'Piéton en roller ou en trotinette', '5':'Autre Véhicule'})
user

Num_Acc place        catu               grav   sexe trajet secu  \
0       201700000001     1  Conducteur  Blessé hospitalsé  homme      9   13   
1       201700000001     2    Passager  Blessé hospitalsé  femme      9   11   
2       201700000001     1  Conducteur  Blessé hospitalsé  homme      1   13   
3       201700000002     1  Conducteur            Indemne  homme      0   11   
4       201700000002     1  Conducteur  Blessé hospitalsé  homme      5   22   
...              ...   ...         ...                ...    ...    ...  ...   
136016  201700060699     1  Conducteur            Indemne  femme      9   11   
136017  201700060700     1  Conducteur            Indemne  femme      9   11   
136018  201700060700     1  Conducteur       Blessé léger  homme      9   21   
136019  201700060700     2    Passager       Blessé léger  femme      9   21   
136020  201700060701     1  Conducteur       Blessé léger  homme      1   21   

       locp actp etatp an_nais num_veh  
0         0    0     0    1968     B01  
1         0    0     0    1973     B01  
2         0    0     0    1967     A01  
3         0    0     0    1953     A01  
4         0    0     0    1960     B01  
...     ...  ...   ...     ...     ...  
136016    0    0     0    1974     B01  
136017    0    0     0    1987     A01  
136018    0    0     0    1991     B01  
136019    0    0     0    1990     B01  
136020    0    0     0    1992     A01  

[136021 rows x 12 columns]

 <li>Afficher le nombre d'occurence de chaque valeur dans lacolonne catu</li>

In [12]:
user['catu'].value_counts()

Conducteur                           100534
Passager                              23685
Piéton                                11509
Piéton en roller ou en trotinette       293
Name: catu, dtype: int64

 


   <li>Combien en moyenne d'accidents par jour ? par mois?</li>
       

In [13]:
#convertir la colonne d'objet à float pour avoir de faire des calculs:
caracter['Num_Acc']= caracter['Num_Acc'].astype(float)


In [14]:
avg_acc_month= caracter.groupby(by= 'mois')['Num_Acc'].count().mean() #==>d'abord il compte combien d'accident par mois il y a, après il fait la moyenne
avg_acc_month


5058.416666666667

In [15]:
avg_acc_day= caracter.groupby(by= 'jour')['Num_Acc'].count().mean()
avg_acc_day


1958.0967741935483

 <li>Combien en moyenne de decès par jour ?</li>

In [16]:
#afin de pouvoir emerger les dataframes il que la colonne cumune a le même type, donc on converti la colonne de Num_Acc à float
user['Num_Acc']= user['Num_Acc'].astype(float)
car_user= pd.merge(caracter, user, on= 'Num_Acc', how= 'inner')
car_user

Num_Acc  an mois jour  hrmn lum agg int atm col  ...        catu  \
0       2.017000e+11  17    1   11  1820   5   2   1   1   1  ...  Conducteur   
1       2.017000e+11  17    1   11  1820   5   2   1   1   1  ...    Passager   
2       2.017000e+11  17    1   11  1820   5   2   1   1   1  ...  Conducteur   
3       2.017000e+11  17    2   13  1630   1   2   3   1   3  ...  Conducteur   
4       2.017000e+11  17    2   13  1630   1   2   3   1   3  ...  Conducteur   
...              ...  ..  ...  ...   ...  ..  ..  ..  ..  ..  ...         ...   
136016  2.017001e+11  17   11   26  1745   2   2   1   1   3  ...  Conducteur   
136017  2.017001e+11  17   11   28  1405   1   2   1   1   3  ...  Conducteur   
136018  2.017001e+11  17   11   28  1405   1   2   1   1   3  ...  Conducteur   
136019  2.017001e+11  17   11   28  1405   1   2   1   1   3  ...    Passager   
136020  2.017001e+11  17   11   30  1939   3   2   1   1   6  ...  Conducteur   

                     grav   sexe trajet secu locp actp etatp an_nais num_veh  
0       Blessé hospitalsé  homme      9   13    0    0     0    1968     B01  
1       Blessé hospitalsé  femme      9   11    0    0     0    1973     B01  
2       Blessé hospitalsé  homme      1   13    0    0     0    1967     A01  
3                 Indemne  homme      0   11    0    0     0    1953     A01  
4       Blessé hospitalsé  homme      5   22    0    0     0    1960     B01  
...                   ...    ...    ...  ...  ...  ...   ...     ...     ...  
136016            Indemne  femme      9   11    0    0     0    1974     B01  
136017            Indemne  femme      9   11    0    0     0    1987     A01  
136018       Blessé léger  homme      9   21    0    0     0    1991     B01  
136019       Blessé léger  femme      9   21    0    0     0    1990     B01  
136020       Blessé léger  homme      1   21    0    0     0    1992     A01  

[136021 rows x 27 columns]

In [17]:
#moyenne de deces par jour
deces = car_user.loc[(car_user['grav'] == 'Tué')]
deces.groupby(by= ['an','mois', 'jour']).count().mean()['Num_Acc']

9.863013698630137

  <li>Trouver les femmes qui ont eu des accidents de type "Indemne" ou "Blessé léger" mais pas plus grave </li>

In [39]:
user.loc[(user['sexe'] == 'femme') & ((user['grav'] == 'Indemne') | (user['grav'] == 'Blessé léger')) , ['grav', 'sexe']]

grav   sexe
8            Indemne  femme
15           Indemne  femme
40           Indemne  femme
41      Blessé léger  femme
44           Indemne  femme
...              ...    ...
136010  Blessé léger  femme
136013  Blessé léger  femme
136016       Indemne  femme
136017       Indemne  femme
136019  Blessé léger  femme

[33993 rows x 2 columns]

 <li>Trouver le nombre d'accidents par sexe et gravité</li> 

In [19]:
user.groupby(['grav', 'sexe'])[['Num_Acc']].count()

Num_Acc
grav              sexe          
Blessé hospitalsé femme     9145
                  homme    19848
Blessé léger      femme    18268
                  homme    28890
Indemne           femme    15725
                  homme    40545
Tué               femme      799
                  homme     2801

 <li>Fusionner usagers2017 et caracteritique2017 sur la colonne Num_Acc </li> 
   
      

- Y a-t-il une nouvelle valeur dans Num_Acc dans la table fusionnée finale qui n'est pas dans l'une des tables ?
    - Non, comme c'est une colonne en commun et en difusionant les 2 dataframes, on prend que les valeurs partagées.

In [20]:
len(user)
len(caracter)

60701

In [21]:
car_user= pd.merge(caracter, user, on= 'Num_Acc', how= 'inner')
car_user.info()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 136021 entries, 0 to 136020
Data columns (total 27 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   Num_Acc  136021 non-null  float64
 1   an       136021 non-null  object 
 2   mois     136021 non-null  object 
 3   jour     136021 non-null  object 
 4   hrmn     136021 non-null  object 
 5   lum      136021 non-null  object 
 6   agg      136021 non-null  object 
 7   int      136021 non-null  object 
 8   atm      135997 non-null  object 
 9   col      136010 non-null  object 
 10  com      136021 non-null  object 
 11  adr      134054 non-null  object 
 12  gps      125823 non-null  object 
 13  lat      117993 non-null  object 
 14  long     117993 non-null  object 
 15  dep      136021 non-null  object 
 16  place    124219 non-null  object 
 17  catu     136021 non-null  object 
 18  grav     136021 non-null  object 
 19  sexe     136021 non-null  object 
 20  trajet   136010 non-null  

  <li>Combien de valeurs manquantes dans la table caracteristiques2017 ?</li>

In [22]:
car_user.isnull().sum().sum() #nombre de valeurs manquantes dans tout la dataframe

69211

   <li>Filtrer les résultats uniquement en prenant les colonnes ayant plus de 0 valeurs manquantes,  trier les par nombre décroissant de valeurs manquantes.</li>

In [23]:
nb_val_manq= car_user[car_user.columns[car_user.isnull().any()]].isnull().sum()
nb_val_manq.sort_values(ascending = False)

lat        18028
long       18028
place      11802
gps        10198
secu        8950
adr         1967
etatp         66
locp          46
actp          43
an_nais       37
atm           24
col           11
trajet        11
dtype: int64

   <li>Afficher le même nombre de valeurs manquantes en pourcentage</li>

In [24]:
len(car_user)

136021

In [25]:
pourcentage= nb_val_manq.sort_values(ascending = False)/len(car_user)
pourcentage

lat        0.132538
long       0.132538
place      0.086766
gps        0.074974
secu       0.065799
adr        0.014461
etatp      0.000485
locp       0.000338
actp       0.000316
an_nais    0.000272
atm        0.000176
col        0.000081
trajet     0.000081
dtype: float64

  <li>Quels sont les accidents survenus en région parisienne (département <=> « 750 » dans le tableau)</li>

In [26]:
paris= car_user.loc[car_user['dep']=='750', ['grav', 'Num_Acc','dep']]
#car_user.groupby(['dep', 'grav'])[['Num_Acc']]
#car_user[['grav','Num_Acc']].groupby([car_user['dep']=='750'])
paris


grav       Num_Acc  dep
103420       Blessé léger  2.017000e+11  750
103421       Blessé léger  2.017000e+11  750
103422       Blessé léger  2.017000e+11  750
103497            Indemne  2.017000e+11  750
103498            Indemne  2.017000e+11  750
...                   ...           ...  ...
131190            Indemne  2.017001e+11  750
131191       Blessé léger  2.017001e+11  750
131192            Indemne  2.017001e+11  750
131193       Blessé léger  2.017001e+11  750
131194  Blessé hospitalsé  2.017001e+11  750

[12034 rows x 3 columns]

 <li>Où se concentrent les accidents les plus graves?</li>

In [27]:
a= car_user.loc[car_user['grav']=='Blessé hospitalsé', ['grav','dep']].value_counts()
b= car_user.loc[car_user['grav']=='Tué', ['grav','dep']].value_counts()
print(a)
print(b)

grav               dep
Blessé hospitalsé  130    1644
                   690    1032
                   930     999
                   590     798
                   060     774
                          ... 
                   230      66
                   976      64
                   550      58
                   480      54
                   900      51
Length: 101, dtype: int64
grav  dep
Tué   130    122
      590     82
      770     81
      340     79
      330     73
            ... 
      650      9
      460      9
      900      8
      480      8
      976      8
Length: 101, dtype: int64


In [28]:
car_user.loc[(car_user['grav']=='Blessé hospitalsé') | (car_user['grav']=='Tué') , ['grav','dep']].groupby('dep').count().max()
car_user.loc[(car_user['grav']=='Blessé hospitalsé') | (car_user['grav']=='Tué') , ['grav','dep']].groupby('dep').count().idxmax()

grav    130
dtype: object

 <li>Renommer les colonnes : jour par day, mois par month et an par year</li>

In [29]:
car_user= car_user.rename(columns = {'jour': 'day', 'mois': 'month', 'an': 'year'})
car_user


Num_Acc year month day  hrmn lum agg int atm col  ...  \
0       2.017000e+11   17     1  11  1820   5   2   1   1   1  ...   
1       2.017000e+11   17     1  11  1820   5   2   1   1   1  ...   
2       2.017000e+11   17     1  11  1820   5   2   1   1   1  ...   
3       2.017000e+11   17     2  13  1630   1   2   3   1   3  ...   
4       2.017000e+11   17     2  13  1630   1   2   3   1   3  ...   
...              ...  ...   ...  ..   ...  ..  ..  ..  ..  ..  ...   
136016  2.017001e+11   17    11  26  1745   2   2   1   1   3  ...   
136017  2.017001e+11   17    11  28  1405   1   2   1   1   3  ...   
136018  2.017001e+11   17    11  28  1405   1   2   1   1   3  ...   
136019  2.017001e+11   17    11  28  1405   1   2   1   1   3  ...   
136020  2.017001e+11   17    11  30  1939   3   2   1   1   6  ...   

              catu               grav   sexe trajet secu locp actp etatp  \
0       Conducteur  Blessé hospitalsé  homme      9   13    0    0     0   
1         Passager  Blessé hospitalsé  femme      9   11    0    0     0   
2       Conducteur  Blessé hospitalsé  homme      1   13    0    0     0   
3       Conducteur            Indemne  homme      0   11    0    0     0   
4       Conducteur  Blessé hospitalsé  homme      5   22    0    0     0   
...            ...                ...    ...    ...  ...  ...  ...   ...   
136016  Conducteur            Indemne  femme      9   11    0    0     0   
136017  Conducteur            Indemne  femme      9   11    0    0     0   
136018  Conducteur       Blessé léger  homme      9   21    0    0     0   
136019    Passager       Blessé léger  femme      9   21    0    0     0   
136020  Conducteur       Blessé léger  homme      1   21    0    0     0   

       an_nais num_veh  
0         1968     B01  
1         1973     B01  
2         1967     A01  
3         1953     A01  
4         1960     B01  
...        ...     ...  
136016    1974     B01  
136017    1987     A01  
136018    1991     B01  
136019    1990     B01  
136020    1992     A01  

[136021 rows x 27 columns]

   <li>Ajoutez « 20 » à chaque élément dans la colonne de l'année récemment renommée (par exemple : 2017) afin d'être un format d'année compréhensible par pd.to_datetime</li>

In [30]:
car_user['year']= car_user['year'].apply(lambda x:"20"+ str(x))
car_user


Num_Acc  year month day  hrmn lum agg int atm col  ...  \
0       2.017000e+11  2017     1  11  1820   5   2   1   1   1  ...   
1       2.017000e+11  2017     1  11  1820   5   2   1   1   1  ...   
2       2.017000e+11  2017     1  11  1820   5   2   1   1   1  ...   
3       2.017000e+11  2017     2  13  1630   1   2   3   1   3  ...   
4       2.017000e+11  2017     2  13  1630   1   2   3   1   3  ...   
...              ...   ...   ...  ..   ...  ..  ..  ..  ..  ..  ...   
136016  2.017001e+11  2017    11  26  1745   2   2   1   1   3  ...   
136017  2.017001e+11  2017    11  28  1405   1   2   1   1   3  ...   
136018  2.017001e+11  2017    11  28  1405   1   2   1   1   3  ...   
136019  2.017001e+11  2017    11  28  1405   1   2   1   1   3  ...   
136020  2.017001e+11  2017    11  30  1939   3   2   1   1   6  ...   

              catu               grav   sexe trajet secu locp actp etatp  \
0       Conducteur  Blessé hospitalsé  homme      9   13    0    0     0   
1         Passager  Blessé hospitalsé  femme      9   11    0    0     0   
2       Conducteur  Blessé hospitalsé  homme      1   13    0    0     0   
3       Conducteur            Indemne  homme      0   11    0    0     0   
4       Conducteur  Blessé hospitalsé  homme      5   22    0    0     0   
...            ...                ...    ...    ...  ...  ...  ...   ...   
136016  Conducteur            Indemne  femme      9   11    0    0     0   
136017  Conducteur            Indemne  femme      9   11    0    0     0   
136018  Conducteur       Blessé léger  homme      9   21    0    0     0   
136019    Passager       Blessé léger  femme      9   21    0    0     0   
136020  Conducteur       Blessé léger  homme      1   21    0    0     0   

       an_nais num_veh  
0         1968     B01  
1         1973     B01  
2         1967     A01  
3         1953     A01  
4         1960     B01  
...        ...     ...  
136016    1974     B01  
136017    1987     A01  
136018    1991     B01  
136019    1990     B01  
136020    1992     A01  

[136021 rows x 27 columns]

   <li>Créer une colonne  date contenant l'année, le mois et le jour</li>
 


In [31]:

datecomplet= car_user['year']+'-'+car_user['month']+'-'+ car_user['day'] #ici on a un type d'objet (utilisant la concatenation)
datecomplet


0          2017-1-11
1          2017-1-11
2          2017-1-11
3          2017-2-13
4          2017-2-13
             ...    
136016    2017-11-26
136017    2017-11-28
136018    2017-11-28
136019    2017-11-28
136020    2017-11-30
Length: 136021, dtype: object

In [32]:
car_user['date']=pd.to_datetime(datecomplet) #ici on a un type de time (utilisant la méthode pd.to_datetime)
car_user['date']


0        2017-01-11
1        2017-01-11
2        2017-01-11
3        2017-02-13
4        2017-02-13
            ...    
136016   2017-11-26
136017   2017-11-28
136018   2017-11-28
136019   2017-11-28
136020   2017-11-30
Name: date, Length: 136021, dtype: datetime64[ns]

In [33]:
car_user

Num_Acc  year month day  hrmn lum agg int atm col  ...  \
0       2.017000e+11  2017     1  11  1820   5   2   1   1   1  ...   
1       2.017000e+11  2017     1  11  1820   5   2   1   1   1  ...   
2       2.017000e+11  2017     1  11  1820   5   2   1   1   1  ...   
3       2.017000e+11  2017     2  13  1630   1   2   3   1   3  ...   
4       2.017000e+11  2017     2  13  1630   1   2   3   1   3  ...   
...              ...   ...   ...  ..   ...  ..  ..  ..  ..  ..  ...   
136016  2.017001e+11  2017    11  26  1745   2   2   1   1   3  ...   
136017  2.017001e+11  2017    11  28  1405   1   2   1   1   3  ...   
136018  2.017001e+11  2017    11  28  1405   1   2   1   1   3  ...   
136019  2.017001e+11  2017    11  28  1405   1   2   1   1   3  ...   
136020  2.017001e+11  2017    11  30  1939   3   2   1   1   6  ...   

                     grav   sexe trajet secu locp actp etatp an_nais num_veh  \
0       Blessé hospitalsé  homme      9   13    0    0     0    1968     B01   
1       Blessé hospitalsé  femme      9   11    0    0     0    1973     B01   
2       Blessé hospitalsé  homme      1   13    0    0     0    1967     A01   
3                 Indemne  homme      0   11    0    0     0    1953     A01   
4       Blessé hospitalsé  homme      5   22    0    0     0    1960     B01   
...                   ...    ...    ...  ...  ...  ...   ...     ...     ...   
136016            Indemne  femme      9   11    0    0     0    1974     B01   
136017            Indemne  femme      9   11    0    0     0    1987     A01   
136018       Blessé léger  homme      9   21    0    0     0    1991     B01   
136019       Blessé léger  femme      9   21    0    0     0    1990     B01   
136020       Blessé léger  homme      1   21    0    0     0    1992     A01   

             date  
0      2017-01-11  
1      2017-01-11  
2      2017-01-11  
3      2017-02-13  
4      2017-02-13  
...           ...  
136016 2017-11-26  
136017 2017-11-28  
136018 2017-11-28  
136019 2017-11-28  
136020 2017-11-30  

[136021 rows x 28 columns]

   <li>Quel est le mois où il y a eu plus d'accidents ?  </li>
    <li>Quel jou où il y a eu plus d'accidents ?</li>
        <li>Quel est le mois où il y a eu plus d'accidents graves pour les femmes?  </li>
   

In [34]:
car_user.groupby(['month'])[['Num_Acc']].count().max()
car_user.groupby(['month'])[['Num_Acc']].count().idxmax()

Num_Acc    6
dtype: object

In [35]:
car_user.groupby(['day'])[['Num_Acc']].count().max() # Num_Acc    4823
car_user.groupby(['day'])[['Num_Acc']].count().idxmax() # 7

Num_Acc    7
dtype: object

In [36]:
car_user.loc[((car_user['grav']=='Blessé hospitalsé') | (car_user['grav']=='Tué')) & (car_user['sexe']== 'femme') , ['grav','sexe','month']].groupby(['month']).count().max()
car_user.loc[((car_user['grav']=='Blessé hospitalsé') | (car_user['grav']=='Tué')) & (car_user['sexe']== 'femme') , ['grav','sexe','month']].groupby(['month']).count().idxmax()


grav    7
sexe    7
dtype: object

 <li>Calculer la distribution quotidienne des décès pendant toute l'année ?</li>

In [37]:
car_user.describe()

Num_Acc
count  1.360210e+05
mean   2.017000e+11
std    1.756535e+04
min    2.017000e+11
25%    2.017000e+11
50%    2.017000e+11
75%    2.017000e+11
max    2.017001e+11